plugins/kotlin/core/src/org/jetbrains/kotlin/idea/core/util/inlineAnalysisUtils.kt

```diff
@@ -124,10 +129,13 @@ private fun analyzeElementWithInline(
 
     analyzedElements.add(element)
 
-    if (declarationsWithBody.isNotEmpty() && deep < 10) {
+    if (declarationsWithBody.isNotEmpty()) {
         for (inlineFunction in declarationsWithBody) {
             val body = inlineFunction.bodyExpression
             if (body != null) {
+                require (deep < TRANSITIVE_INLINE_ANALYSIS_DEPTH_LIMIT) {
```

Comment:
@yanex what is the motivation for this limit?

Reply:
I don't know, this code is quite old and originally came from the bytecode tool window: https://github.com/JetBrains/kotlin/commit/715b9ad859fb70e1c18f9d95a2484b00f6a22d00#diff-2a5f5f2b5b87193d3b3c581964190c1bd577b27ef8beefa85ae1a00e53d5d4c5R315.
I'm totally ok in removing this limitation.